csp-mixin/alerts/gcp-alerts.yml

```diff
@@ -14,3 +14,129 @@ groups:
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLActiveConnections
+        expr: |
+          avg by (job,project_id,instance, database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_threads{thread_kind="THREADS_CONNECTED", job=~".+",project_id=~".+",instance=~".+", database_id=~".+"}) > 0.9 * avg by (job,project_id,instance, database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_max_connections{job=~".+",project_id=~".+",instance=~".+", database_id=~".+"})
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'Too many active connections.'
+          description: 'Investigate connection pooling settings and connection management in your {{ $labels.database_id }} application database.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLAbortedConnections
+        expr: |
+          sum by(job, instance, project_id)(rate(stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_aborted_connects_count[5m])) > 5
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'More than 5 failed connections in 5 minutes.'
+          description: 'Verify credentials and network settings; check for firewall rules blocking connections for the {{ $labels.database_id }} database.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLLagSecondsBehindMaster
+        expr: |
+          avg by (job,project_id,instance, database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_replication_seconds_behind_master) > 5
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: warning
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'More than 5 seconds lag between read replica and primary.'
+          description: 'Check {{ $labels.database_id }} database for network latency between primary and replica; adjust configurations to optimize replication.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpPubSubNumUndeliveredMessages
+        expr: |
+          avg by (job,project_id,instance)(stackdriver_pubsub_subscription_pubsub_googleapis_com_subscription_num_undelivered_messages{job=~".+",project_id=~".+",instance=~".+"}) > 1000
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: warning
+          service: 'Pub/Sub'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'More than 1000 messages.'
```

Comment:
It's not clear to what service is associated. Can you add something that mention the word subscription at least?